src/textual/widgets/_tree.py

```diff
@@ -159,24 +159,60 @@ def allow_expand(self, allow_expand: bool) -> None:
         self._allow_expand = allow_expand
         self._updates += 1
 
-    def expand(self) -> None:
-        """Expand a node (show its children)."""
+    def _expand(self, expand_all: bool) -> None:
+        """Mark a node as expanded (its children are shown).
+
+        Args:
+            expand_all: If `True` expand all offspring at all depths.
+        """
         self._expanded = True
         self._updates += 1
+        if expand_all:
+            for child in self.children:
+                child._expand(expand_all)
+
+    def expand(self, *, expand_all: bool = False) -> None:
```

Comment:
There's some redundancy in the parameter name here `exapand_all`, and friends.

At the moment to expand all nodes you would do this:

```python
node.expand(expand_all=True)
```

Two mentions of "expand". But its not like there is any ambiguity re what the "all" refers to.

How about this?

```python
node.expand(all=True)
```